Add numeric fused head dataset

ghomasHudson opened the pull request:
Adding the [NFH: Numeric Fused Head](https://nlp.biu.ac.il/~lazary/fh/) dataset.

Everything looks sensible and I've included both the identification and resolution tasks. I haven't personally used this dataset in my research so am unable to specify what the default configuration / supervised keys should be.

I've filled out the basic info on the model card to the best of my knowledge but it's a little tricky to understand exactly what the fields represent.

Dataset author: @yanaiela

yanaiela commented on datasets/numeric_fused_head/README.md:
```diff
@@ -0,0 +1,190 @@
+---
+annotations_creators:
+  identification:
+  - expert-generated
```
The identification dataset actually has two with different quality annotation types:
1) the training and development splits were created automatically, using a constituency parser.
2) the test set was curated by experts (the authors)

ghomasHudson replied:
Added!

yanaiela commented on datasets/numeric_fused_head/README.md:
```diff
@@ -0,0 +1,190 @@
+- **Homepage:** [The Numeric Fused-Head demo](https://nlp.biu.ac.il/~lazary/fh/)
+- **Repository:** [Github Repo](https://github.com/yanaiela/num_fh)
+- **Paper:** [Where’s My Head? Definition, Dataset and Models for Numeric Fused-Heads Identification and Resolution](https://www.mitpressjournals.org/doi/full/10.1162/tacl_a_00280)
+- **Leaderboard:**
```
can add the nlpprogress page - http://nlpprogress.com/english/missing_elements.html

ghomasHudson replied:
Added!

yanaiela reviewed (commented):
Thanks for adding this @ghomasHudson! 
I added some comments for some of the fields.

Also, I'm not sure about this since I haven't used the library yet, but maybe it's worth adding the identification and resolution as two separate datasets?

ghomasHudson commented:
> Thanks for adding this @ghomasHudson!
> I added some comments for some of the fields.
> 
> Also, I'm not sure about this since I haven't used the library yet, but maybe it's worth adding the identification and resolution as two separate datasets?

Thanks for replying - I hope this will make your dataset more accessible to a wider audience - I've added the changes to the model card you suggested.

In terms of the identification and resolution tasks, I've currently added them as separate `splits` in huggingface/datasets so you can load identification like this:

```
import datasets
dataset = datasets.load_dataset("numeric_fused_head", "identification")
print(dataset["train"][0])
>> {"tokens": ["The", "quick", "brown", "fox",....] "start_index": 11, "end_index": 12, "label": 0}
```
And resolution like this:

```
import datasets
dataset = datasets.load_dataset("numeric_fused_head", "resolution")
print(dataset["train"][0])
>> {"tokens": ["The", "quick", "brown", "fox",....] "head": ["AGE"], "anchors_indices": [12], ...}
```

yanaiela commented:
I hope so too, thanks!

Re the splits, that makes sense to me.

lhoestq commented on datasets/numeric_fused_head/numeric_fused_head.py:
```diff
@@ -0,0 +1,141 @@
+}
+
+
+class NumericFusedHeads(datasets.GeneratorBasedBuilder):
```
```suggestion
class NumericFusedHead(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `numeric_fused_head`

lhoestq commented on datasets/numeric_fused_head/dataset_infos.json:
```diff
@@ -0,0 +1 @@
+{"identification": {"description": "Fused Head constructions are noun phrases in which the head noun is missing and is said to be \"fused\" with its dependent modifier. This missing information is implicit and is important for sentence understanding.The missing heads are easily filled in by humans,  but pose a challenge for computational models.\n\nFor example, in the sentence: \"I bought 5 apple ...
```
```suggestion
{"identification": {"description": "Fused Head constructions are noun phrases in which the head noun is missing and is said to be \"fused\" with its dependent modifier. This missing information is implicit and is important for sentence understanding.The missing heads are easily filled in by humans,  but pose a challenge for computational models.\n\nFor example, in the sentence: \"I bought 5 apples but got only 4.\", 4 is a Fused-Head, and the missing head is apples, which appear earlier in the sentence.\n\nThis is a crowd-sourced dataset of 10k numerical fused head examples (1M tokens).\n", "citation": "@article{elazar_head,\n    author = {Elazar, Yanai and Goldberg, Yoav},\n    title = {Where\u2019s My Head? Definition, Data Set, and Models for Numeric Fused-Head Identification and Resolution},\n    journal = {Transactions of the Association for Computational Linguistics},\n    volume = {7},\n    number = {},\n    pages = {519-535},\n    year = {2019},\n    doi = {10.1162/tacl\\_a\\_00280},\n    URL = {https://doi.org/10.1162/tacl_a_00280},\n}\n", "homepage": "https://nlp.biu.ac.il/~lazary/fh/", "license": "MIT", "features": {"tokens": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "start_index": {"dtype": "int32", "id": null, "_type": "Value"}, "end_index": {"dtype": "int32", "id": null, "_type": "Value"}, "label": {"num_classes": 2, "names": ["neg", "pos"], "names_file": null, "id": null, "_type": "ClassLabel"}}, "post_processed": null, "supervised_keys": null, "builder_name": "numeric_fused_head", "config_name": "identification", "version": {"version_str": "0.0.0", "description": null, "major": 0, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 22290345, "num_examples": 165606, "dataset_name": "numeric_fused_head"}, "test": {"name": "test", "num_bytes": 68282, "num_examples": 500, "dataset_name": "numeric_fused_head"}, "validation": {"name": "validation", "num_bytes": 2474528, "num_examples": 18401, "dataset_name": "numeric_fused_head"}}, "download_checksums": {"https://raw.githubusercontent.com/yanaiela/num_fh/master/data/identification/processed/train.tsv": {"num_bytes": 21908366, "checksum": "5534fbfdfa2d7b1a370e4a303fcc5ea324f8aae4d5ad0b2601961225c38f19eb"}, "https://raw.githubusercontent.com/yanaiela/num_fh/master/data/identification/processed/test.tsv": {"num_bytes": 67315, "checksum": "4d0f3b3b103ad592b8ad0a3e6d24c28bff6afb684e8bc08d22da8070f0e73c13"}, "https://raw.githubusercontent.com/yanaiela/num_fh/master/data/identification/processed/dev.tsv": {"num_bytes": 2431839, "checksum": "457ebe4342f5c6d53a04c8b49a50bcdee47b3d156aab3e37f8e79ff55823f02a"}}, "download_size": 24407520, "post_processing_size": null, "dataset_size": 24833155, "size_in_bytes": 49240675}, "resolution": {"description": "Fused Head constructions are noun phrases in which the head noun is missing and is said to be \"fused\" with its dependent modifier. This missing information is implicit and is important for sentence understanding.The missing heads are easily filled in by humans,  but pose a challenge for computational models.\n\nFor example, in the sentence: \"I bought 5 apples but got only 4.\", 4 is a Fused-Head, and the missing head is apples, which appear earlier in the sentence.\n\nThis is a crowd-sourced dataset of 10k numerical fused head examples (1M tokens).\n", "citation": "@article{elazar_head,\n    author = {Elazar, Yanai and Goldberg, Yoav},\n    title = {Where\u2019s My Head? Definition, Data Set, and Models for Numeric Fused-Head Identification and Resolution},\n    journal = {Transactions of the Association for Computational Linguistics},\n    volume = {7},\n    number = {},\n    pages = {519-535},\n    year = {2019},\n    doi = {10.1162/tacl\\_a\\_00280},\n    URL = {https://doi.org/10.1162/tacl_a_00280},\n}\n", "homepage": "https://nlp.biu.ac.il/~lazary/fh/", "license": "MIT", "features": {"tokens": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "line_indices": {"feature": {"dtype": "int32", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "head": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "speakers": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "anchors_indices": {"feature": {"dtype": "int32", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}}, "post_processed": null, "supervised_keys": null, "builder_name": "numeric_fused_head", "config_name": "resolution", "version": "0.0.0", "splits": {"train": {"name": "train", "num_bytes": 19766437, "num_examples": 7412, "dataset_name": "numeric_fused_head"}, "test": {"name": "test", "num_bytes": 2743071, "num_examples": 1000, "dataset_name": "numeric_fused_head"}, "validation": {"name": "validation", "num_bytes": 2633549, "num_examples": 1000, "dataset_name": "numeric_fused_head"}}, "download_checksums": {"https://raw.githubusercontent.com/yanaiela/num_fh/master/data/resolution/processed/nfh_train.jsonl": {"num_bytes": 19589828, "checksum": "9391e454c4dbfe2db27f4fe8fb0d1af320e27306e1e6735553d348407cea2b2e"}, "https://raw.githubusercontent.com/yanaiela/num_fh/master/data/resolution/processed/nfh_test.jsonl": {"num_bytes": 2723429, "checksum": "8962497910531544fd162cd212533e590b41165759490d781063b1a67d4caded"}, "https://raw.githubusercontent.com/yanaiela/num_fh/master/data/resolution/processed/nfh_dev.jsonl": {"num_bytes": 2610146, "checksum": "967174c741cb69cb98ac7a091e620f26e766c4a766fb679c45743933a144e095"}}, "download_size": 24923403, "post_processing_size": null, "dataset_size": 25143057, "size_in_bytes": 50066460}}
```

`numeric_fuse_heads` -> `numeric_fuze_head`

lhoestq reviewed (approved):
Awesome thanks for adding this one :)
